### PR TITLE
MVP series labels

### LINKED ISSUE #59: Seaching Series by label
The series should be searchable by label.

Stories:

1. Enter/select label in the filter

### LINKED ISSUE #60: Adding a new label to a series
**AS A** user
**I WANT TO** to create a new label and assigned it to a selected series
**SO THAT** this label remains assigned to the series for future searches

The details of the HOW have to be discussed together

- [ ] In the UI, be able to create a new label
- [ ] Backend: REST API to assign a label to a series. The data model must followed the diagram.

### PR DESCRIPTION
**Pull request by bat3**
First implementation of MVP series label.
Need talk about architecture and use cases

### CODE REVIEW

File: client/web/src/components/series/label/SeriesLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const handleDelete = (localSeries: ISeries | null, labelClicked: ILabel) => {
+    if(localSeries == null)
+    {
+      throw Error("No series selected")
+    }
+    if(localSeries.labels == null)
+    {
+      localSeries.labels = new Array<ILabel>();
+    }
+    localSeries.labels.forEach((label: ILabel) => {
```
Comment (ptitdam2001) [inline client/web/src/components/series/label/SeriesLabel.tsx]: Why do you do a forEach here?

You just need next line to remove label

```typescript
localSeries.label = localSeries.labels.filter((label: ILabel) => label.id !== labelClicked.id)
```

File: client/web/src/components/series/label/SeriesLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      var newLabels = localSeries.labels.filter((label) => { 
+        if(label !== labelClicked){
+          return label;
+        }
+        return null;
+      });
+      localSeries.labels = newLabels;
+    });
+    SeriesService.update(localSeries.id, localSeries).then(()=> {dispatch({ type: actionTypes.FETCH_ALL_SERIES })});
```
Comment (ptitdam2001) [inline client/web/src/components/series/label/SeriesLabel.tsx]: You can use redux-saga here :

In seriesSaga.ts : 
```typescript
function* updateSeries(action) {
  const { series } = action
  yield call([SeriesService, 'update'], series.id, series)
  yield call(fetchSeries)
}
// You have to update actionWatcher function with
yield takeLatest(actionTypes.UPDATE_SERIES, updateSeries) //Add actionType UPDATE_SERIES
}
```
In feature/user/types.ts, you must add :
```typescript
interface UpdateSeriesAction {
  type: typeof UPDATE_SERIES
  series: ISeries
}

// Add to type SeriesActionTypes
```

finally, you just can call in that file the line below:
```typescript
dispatch({ type: UPDATE_SERIES, series: localSeries
```

File: client/web/src/components/series/label/SeriesLabelFrom.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          if(props.series == null)
+          {
+            throw Error("No series selected")
+          }
+          if(props.series.labels == null)
+          {
+            props.series.labels = new Array<ILabel>();
+          }
+          props.series.labels.push(values);
+          SeriesService.update(props.series.id, props.series).then(()=> {dispatch({ type: actionTypes.FETCH_ALL_SERIES })});
```
Comment (ptitdam2001) [inline client/web/src/components/series/label/SeriesLabelFrom.tsx]: Following previous comment, you can change with: 

```typescript
dispatch({type: UPDATE_SERIES, series: props.series})
```
But You should assign props.series to one variable, to be sure there is no side effect

File: client/web/src/models/ILabel.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export interface ILabel {
+  "id": string,
```
Comment (ptitdam2001) [inline client/web/src/models/ILabel.ts]: you remove double-quote and virgule in entire file

File: client/web/src/models/series.ts [hunk 5]
```diff
@@ -6,4 +8,5 @@ export interface ISeries {
   numberOfSeriesRelatedInstances: string
   patientsName: string
   bodyPartExamined: string
+  labels: Array<ILabel>
```
Comment (ptitdam2001) [inline client/web/src/models/series.ts:11]: the type can be : ILabel[]

**Review by ptitdam2001 — CHANGES_REQUESTED** [review]

File: client/web/src/components/series/label/SeriesLabel.tsx (hunk 1 above)
Comment (bat3) [reply client/web/src/components/series/label/SeriesLabel.tsx]: Done

File: client/web/src/components/series/label/SeriesLabel.tsx (hunk 2 above)
Comment (bat3) [reply client/web/src/components/series/label/SeriesLabel.tsx]: Done

File: client/web/src/models/ILabel.ts (hunk 4 above)
Comment (bat3) [reply client/web/src/models/ILabel.ts]: Done

File: client/web/src/components/series/label/SeriesLabelFrom.tsx (hunk 3 above)
Comment (bat3) [reply client/web/src/components/series/label/SeriesLabelFrom.tsx]: Done

**Review by cquennouelle — APPROVED** [review]

**Review by ptitdam2001 — APPROVED** [review]

**Pull request merged**